src/transport.rs

```diff
@@ -0,0 +1,61 @@
+/********************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use async_trait::async_trait;
+
+use up_rust::{
+    transport::datamodel::UTransport,
+    uprotocol::{UMessage, UStatus, UUri},
+    uuid::builder::UUIDBuilder,
+};
+
+pub struct MqttTransport {}
```

Comment:
Is it intended to be that a single "object" could be used for both doing `uP-L1` operations (e.g. `UTransport::send()` as well as `uP-L2` operations (e.g. `RpcClient::invoke_method()`)?

That's what's being done [over here](https://github.com/eclipse-uprotocol/up-client-zenoh-rust/pull/2/files#r1506634198) for the Zenoh implementation, but I'm not clear on the aim of this `up-client` library.

Or is it intentional to have there be a strict delineation between `MqttTransport` and any future `MqttRpcServer` and `MqttRpcClient`?

Reply:
I have updated the project to change the object to "UPClientMqtt" and placed that object in the lib.rs file, similar to how it was done in the Zenoh implementation.